File: sparkstream/src/main/scala/com/gyt/sparkstream/project/dao/CourseClickCountDAO.scala
package com.gyt.sparkstream.project.dao

import com.gyt.sparkstream.project.domain.CourseClickCount
import com.gyt.sparkstream.project.utils.HBaseUtils
import org.apache.hadoop.hbase.client.Get
import org.apache.hadoop.hbase.util.Bytes

import scala.collection.mutable.ListBuffer

/**
  * 实战课程数据库访问层
  */
object CourseClickCountDAO {
  val tableName = "course_clickcount"
  val cf = "info"
  val qualifer = "click_count"

  /**
    * 保存数据到Hbase
    * @param list CourseClickCount集合
    */
  def save(list: ListBuffer[CourseClickCount]): Unit = {
    val table = HBaseUtils.getInstance().getTable(tableName)

    for (elem <- list) {
      // incrementColumnValue可以把相同行键到值相加
      table.incrementColumnValue(Bytes.toBytes(elem.day_courseId), Bytes.toBytes(cf),
        Bytes.toBytes(qualifer), elem.click_count, false)
    }

  }

  /**
    * 根据rowkey查询
    * @param day_courseId rowkey
    * @return
    */
  def query(day_courseId: String): Long = {
    val table = HBaseUtils.getInstance().getTable(tableName)

    val get = new Get(Bytes.toBytes(day_courseId))
    val value = table.get(get).getValue(cf.getBytes, qualifer.getBytes)

    if (value = null) {
      0L
    } else {
      Bytes.toLong(value)
    }
  }


}

File: sparkstream_web/src/main/java/com/gyt/dao/CourseClickCountDAO.java
package com.gyt.dao;


import com.gyt.domain.CourseClickCount;
import com.gyt.utils.HBaseUtils;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * 实战课程访问数量访问层
 */

@Component
public class CourseClickCountDAO {

    public List<CourseClickCount> query(String day) throws Exception {
        List<CourseClickCount> list = new ArrayList<>();

        // 到HBase中根据day获取数据
        Map<String, Long> map = HBaseUtils.getInstance().query("course_clickcount", day);

        for (Map.Entry<String, Long> entry : map.entrySet()) {
            CourseClickCount model = new CourseClickCount();
            model.setName(entry.getKey());
            model.setValue(entry.getValue());
            list.add(model);
        }

        return list;
    }

}
